<center><h1 style="color:green">Exercise</h1></center>
Use iris flower dataset from sklearn library and use cross_val_score against following models to measure the performance of each. In the end figure out the model with best performance,

1. Logistic Regression
2. SVM
3. Decision Tree
4. Random Forest

In [1]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [2]:
iris = load_iris()

<b>Logistic Regression

In [3]:
l_scores = cross_val_score(LogisticRegression(max_iter=500), iris.data, iris.target,cv=3)
l_scores

array([0.98, 0.96, 0.98])

In [4]:
np.average(l_scores)

0.9733333333333333

<b>Decision Tree

In [5]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target,cv=3)
d_scores

array([0.98, 0.92, 0.98])

In [6]:
np.average(d_scores)

0.96

<b>Support Vector Machine (SVM)

In [7]:
s_scores = cross_val_score(SVC(), iris.data, iris.target,cv=3)
s_scores

array([0.96, 0.98, 0.94])

In [8]:
np.average(s_scores)

0.96

<b>Random Forest

In [9]:
r_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target,cv=3)
r_scores

array([0.98, 0.94, 0.96])

In [10]:
np.average(r_scores)

0.96

<b>Best score so far is from Logistic Regression: 0.9733333333333333